core/chain_util.go

```diff
@@ -346,3 +349,34 @@ func GetBlockByHashOld(db ethdb.Database, hash common.Hash) *types.Block {
 	}
 	return (*types.Block)(&block)
 }
+
+// returns a formatted MIP mapped key.
+func mipmapKey(num, level uint64) []byte {
+	key := new(big.Int).SetUint64(num / level * level)
+
+	return append(mipmapPre, key.Bytes()...)
+}
```

Comment:
Because of saturation we have the different levels. That's the purpose of it. Quickly search of ranges giving you a probability of inclusion. 

The keys contain the ranges. No need for level